File: languages/go/entity/identity/integer_attribute.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IntegerAttribute represents an attribute with an integer value.
type IntegerAttribute struct {
	Attribute
}

// NewIntegerAttribute creates a new IntegerAttribute instance with the provided key and value.
func NewIntegerAttribute(key string, value int64) IntegerAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return IntegerAttribute{
		Attribute: a,
	}
}

func NewIntegerAttributeFromProto(s *proto.IntegerAttribute) IntegerAttribute {
	if s == nil {
		return IntegerAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return IntegerAttribute{
		Attribute: a,
	}
}

func (s IntegerAttribute) ToProto() *proto.IntegerAttribute {
	return &proto.IntegerAttribute{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(int64),
	}
}

File: languages/js/src/entity/identity/integer_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/bloock_identity_entities";
import { Attribute } from "./attribute";

/**
 * Represents an attribute with an integer value.
 */
export class IntegerAttribute extends Attribute<number> {
  public toProto(): identityEntitiesProto.IntegerAttribute {
    return identityEntitiesProto.IntegerAttribute.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(
    r: identityEntitiesProto.IntegerAttribute
  ): IntegerAttribute {
    return new IntegerAttribute(r.id, r.value);
  }
}
